src/cli/glue_gen.rs

```diff
@@ -27,22 +27,39 @@ impl GlueCrateGenerator {
         self.add_glue_code(temp_path.as_path())
             .context("Failed to add glue code")?;
 
+        // create a gluecrate with the temp_path and that can access the manifest path
```

Comment:
That's it. I think the best solution here is to store the [`TempDir`] that you generated in `create_temp_dir` in the `GlueCrate`, so that it does not get dropped at the end of the function.

[`TempDir`]: https://docs.rs/tempfile/3.2.0/tempfile/struct.TempDir.html

Reply:
spoiler alert : 
![image](https://user-images.githubusercontent.com/13097851/126074971-0674778c-c7bc-4e6f-be0b-ae69bf2e46a0.png)